server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1046,6 +1049,7 @@ private void doDeleteShardSnapshots(
                 repositoryStateId,
                 repoMetaVersion,
                 Function.identity(),
+                Priority.NORMAL,
```

Comment:
this is the existing behavior. Since we are introducing a new parameter to help modify the priority for v2 snapshots as immediate, need to explicitly mention this.

Reply:
Sorry what is the concern with NORMAL here? Is this in the sync API call path?

Reply:
We have made cluster state update task for repository metadata update as `IMMEDIATE` for create snapshot v2 (in sync path). To ensure that the changes are done only for snapshot v2 we have to pass this parameter in the method signature. The default priority for cluster state update task is `NORMAL`, so passing `NORMAL` at other places where same method is getting called.

Reply:
Sorry why is this cluster state update needed at all given that we are not moving shards snapshots in-progress. Is my understanding correct?

Reply:
We do need this for backward compatibility now. The repository metadata is read from cluster state at a number of places.